src/modules/ModuleWithItemsFromConfig.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace WPML\PB\Cornerstone\Modules;
+
+use WPML\FP\Obj;
+
+class ModuleWithItemsFromConfig extends \WPML_Cornerstone_Module_With_Items {
+
+	/** @var array $fieldDefinitions */
+	private $fieldDefinitions = [];
+
+	public function __construct( array $config ) {
+		foreach ( $config as $fieldConfig ) {
+			$field                            = Obj::prop( 'field', $fieldConfig );
+			$this->fieldDefinitions[ $field ] = $fieldConfig;
+		}
+	}
+
+	private function getFieldData( $field, $key ) {
+		return Obj::path( [ $field, $key ], $this->fieldDefinitions );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_title( $field ) {
+		return $this->getFieldData( $field, 'type' );
```

Comment:
@strategio Do we really need the `getFieldData` helper? You could replace this call with:
```php
return Obj::path( [ $field, 'type' ], $this->fieldDefinitions );
```

Reply:
Right, changed.